### DISCUSSION TITLE
Help with testing

### DISCUSSION THREAD

**Discussion by joshfester** [bensheldon/good_job#1153] [Q&A] [answered]
I'm migrating to good_job from sidekiq (and switching to solid_cache, bye redis!). With sidekiq, I liked to test the number of jobs, then execute the jobs and test the results. It would look something like this:

```
Model.perform_some_action

assert_equal 1, TheJob.size

Sidekiq::Worker.drain_all

assert_results
```

So I have two questions:

1. How do I check the number of jobs in the queue for a give job?
2. How do I execute all the jobs?

Thanks so much!

Update: I found `GoodJob::Job.queued.count`, looks like I should use that to check the queue size?

**Comment by bensheldon** [comment, marked as answer]
Now that you're using ActiveJob, I would recommend that you do unit testing directly with Active Job's test adapter (which sidesteps GoodJob): https://edgeguides.rubyonrails.org/testing.html#testing-jobs-in-context

For integration testing, you can use GoodJob with `:external` execution mode, which will cause all jobs to go to the database. Something like:


```ruby
ActiveJob::Base.queue_adapter = GoodJob::Adapter.new(execution_mode: :external)

MyJob.perform_later

assert_equal 1, GoodJob::Job.where(job_class: "MyJob").count # this is an Active Record model / scope chain

GoodJob.perform_inline

assert_results

```

**Reply by joshfester** [reply]
Ahh thank you so much!! This is perfect 🙏
